### PR TITLE
Add target-params to Platform Tuple

### PR DESCRIPTION
**Pull request by sdpenguin**
Added support for specifying target parameters in the Platform tuple.

### CODE REVIEW

**Comment by sdpenguin** [comment]
Have updated the commit to remove JUNO_SHELL_PROMPT reference and make it compatible with the current tip of next.

**Comment by sdpenguin** [comment]
Updated to stop the overwrite of existing target_params.

File: wa/framework/target/descriptor.py [hunk 1]
```diff
@@ -416,26 +416,30 @@ def get_descriptions(self):
     'chromeos': ChromeOsAssistant
 }
 
-# name --> ((platform_class, conn_class), params_list, defaults)
+# name --> (((platform_class, conn_class), params_list, defaults), target_defaults)
 # Note: normally, connection is defined by the Target name, but
 #       platforms may choose to override it
+# Note: the target_defaults allows you to override common target_params for a
+# particular platform. Parameters you can override are in COMMON_TARGET_PARAMS
+import re
```
Comment (marcbonnici) [inline wa/framework/target/descriptor.py]: This should not be imported here. Is it needed at all?

File: wa/framework/target/descriptor.py [hunk 2]
```diff
@@ -416,26 +416,30 @@ def get_descriptions(self):
...
+    'generic': ((Platform, None), COMMON_PLATFORM_PARAMS, None, None),
     'juno': ((Juno, None), COMMON_PLATFORM_PARAMS + VEXPRESS_PLATFORM_PARAMS,
             {
                  'vemsd_mount': '/media/JUNO',
                  'baudrate': 115200,
                  'bootloader': 'u-boot',
                  'hard_reset_method': 'dtr',
-            }),
+            },
+            {'shell_prompt': DEFAULT_SHELL_PROMPT},),
```
Comment (marcbonnici) [inline wa/framework/target/descriptor.py]: Do we need to override ``shell_prompt`` with its default value?

**Comment by sdpenguin** [comment]
Have updated the code to stop the redundant override, and added an example of how to override in the comments.

File: wa/framework/target/descriptor.py (hunk 1 above)
Comment (sdpenguin) [reply wa/framework/target/descriptor.py]: Sorry. Whoops! Have removed it.

**Pull request merged**